tests/sklearn/test_sklearn_autolog.py

```diff
@@ -1060,6 +1060,25 @@ def test_autolog_does_not_throw_when_infer_signature_fails():
     assert "signature" not in model_conf.to_dict()
 
 
+def test_autolog_does_not_warn_when_model_has_transform_function():
+    X, y = get_iris()
+
+    with mlflow.start_run() as run, mock.patch("mlflow.sklearn._logger.warning") as mock_warning:
+        mlflow.sklearn.autolog(log_input_examples=True, log_model_signatures=True)
+        estimators = [
+            ("std_scaler", sklearn.preprocessing.StandardScaler()),
+        ]
+        model = sklearn.pipeline.Pipeline(estimators)
+        model.fit(X, y)
+
+    # Warning not called
+    with pytest.raises(AssertionError, match=r"expected call not found"):
+        mock_warning.assert_called_with("Failed to infer model signature: Failed")
+
+    model_conf = get_model_conf(run.info.artifact_uri)
+    assert "signature" in model_conf.to_dict(), model_conf.to_dict()
```

Comment:
```suggestion
    assert "signature" in model_conf.to_dict()
```

`model_conf.to_dict()` is not needed. pytest should tell us what it is if this test fails on this line.